tests/Doctrine/Tests/Models/GH7505/TextResponse.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\Models\GH7505;
+
+use Doctrine\ORM\Annotation as ORM;
+
+/**
+ * @ORM\Entity()
+ */
+class TextResponse extends AbstractResponse
+{
+    /**
+     * @ORM\Column(name="value_string", type="string")
```

Comment:
In order to avoid confusing this issue with what discussed in the original thread, please use the same `type` for both entities here, so that we first verify only the hydration issues, and not the schema validator ones.

Reply:
Hello,

I don't quite understand what you mean: if both columns have the same type, the problem doesn't happen.

Reply:
As discussed in https://github.com/doctrine/doctrine2/issues/7505#issuecomment-444924119, the test scenarios are two:

 * one with failed schema validation (same column name in STI, but with different types)
 * one with failed hydration (this patch - different column, same reflection property, different types)

I think this particular discussion point can therefore be dropped.